engine/src/flutter/shell/platform/android/io/flutter/embedding/engine/FlutterEngineCommandLineFlags.java

```diff
@@ -0,0 +1,243 @@
+  public static final String USE_TEST_FONTS = "--use-test-fonts";
+
+  /** Sets the port for the Dart VM Service. */
+  public static final String VM_SERVICE_PORT = "--vm-service-port=";
+
+  /** Enables Vulkan validation layers if available. */
+  public static final String ENABLE_VULKAN_VALIDATION = "--enable-vulkan-validation";
+
+  /** Enables GPU tracing for OpenGL. */
+  public static final String ENABLE_OPENGL_GPU_TRACING = "--enable-opengl-gpu-tracing";
+
+  /** Enables GPU tracing for Vulkan. */
+  public static final String ENABLE_VULKAN_GPU_TRACING = "--enable-vulkan-gpu-tracing";
+
+  /**
+   * Set whether leave or clean up the VM after the last shell shuts down. It can be set from app's
+   * meta-data in the application block in AndroidManifest.xml. Set it to true in to leave the Dart
+   * VM, set it to false to destroy VM.
+   *
+   * <p>If your want to let your app destroy the last shell and re-create shells more quickly, set
+   * it to true, otherwise if you want to clean up the memory of the leak VM, set it to false.
+   *
+   * <p>TODO(eggfly): Should it be set to false by default?
+   * https://github.com/flutter/flutter/issues/96843
+   */
+  public static final String LEAK_VM = "--leak-vm=";
+
+  public static final List<String> ALL_FLAGS =
+      Collections.unmodifiableList(
+          Arrays.asList(
+              TRACE_STARTUP,
+              START_PAUSED,
+              DISABLE_SERVICE_AUTH_CODES,
+              ENDLESS_TRACE_BUFFER,
+              ENABLE_DART_PROFILING,
+              PROFILE_STARTUP,
+              TRACE_SKIA,
+              TRACE_SKIA_ALLOWLIST,
+              TRACE_SYSTRACE,
+              TRACE_TO_FILE,
+              PROFILE_MICROTASKS,
+              DUMP_SKP_ON_SHADER_COMPILATION,
+              PURGE_PERSISTENT_CACHE,
+              VERBOSE_LOGGING,
+              DART_FLAGS,
+              ENABLE_SOFTWARE_RENDERING,
+              SKIA_DETERMINISTIC_RENDERING,
+              AOT_SHARED_LIBRARY_NAME,
+              FLUTTER_ASSETS_DIR,
+              OLD_GEN_HEAP_SIZE,
+              ENABLE_IMPELLER,
+              IMPELLER_BACKEND,
+              DISABLE_MERGED_PLATFORM_UI_THREAD,
+              ENABLE_SURFACE_CONTROL,
+              ENABLE_FLUTTER_GPU,
+              IMPELLER_LAZY_SHADER_MODE,
+              IMPELLER_ANTIALIAS_LINES,
+              VM_SNAPSHOT_DATA,
+              ISOLATE_SNAPSHOT_DATA,
+              USE_TEST_FONTS,
+              VM_SERVICE_PORT,
+              ENABLE_VULKAN_VALIDATION,
+              ENABLE_OPENGL_GPU_TRACING,
+              ENABLE_VULKAN_GPU_TRACING,
+              LEAK_VM));
+
+  /**
+   * Converts a command line flag string to camel case that can be used as a manifest metadata name
+   * if the flag is configurable by manifest. See {@link FlutterEngineManifestFlags} for the list of
+   * flags for such available flags.
+   *
+   * <p>Examples: " --old-gen-heap-size=" becomes "oldGenHeapSize" and "--use-test-fonts" becomes
+   * "useTestFonts".
```

Comment:
Looking for feedback on the naming. I know camel case is idiomatic to Android and this is a pretty simple method that does that but I realize that using a conversion method is funky. We could also hard code the manifest names.